Computer History Museum Archive.js

```diff
@@ -0,0 +1,589 @@
+					snapshot: false
+				});
+			}
+		})
+
+	}
+
+	newItem.complete()
+}
+
+
+// helper functions
+function insertToTheStartOfAbstract(insert, abstract, title) {
+	const abstractContent = title + ': ' + insert + '\n' + (abstract ? abstract : '')
+	return abstractContent  ? abstractContent : ''
+}
+function getContent(node, fieldTitle) {
+	const content = node.textContent.replace(fieldTitle, '').trim()
+	return content ? content : ''
+}
+function addToExtra(oldExtra, newContent, fieldTitle) {
+	return (oldExtra ? oldExtra + '\n' : '') + fieldTitle + ': ' + newContent.replace(/  /g, '').replace(/\n\n/g, '').replace('\n', ': ')
+}
+
+function getContributors(node) {
+	let contributors = []
+	const people = node.querySelectorAll('td')
+	people.forEach(personDiv => {
+		const spans = personDiv.querySelectorAll('span')
+		const name = spans[0].textContent
+		const [lastName, firstName] = name.split(', ')
+		Zotero.debug(lastName)
+		Zotero.debug(firstName)
+		contributors.push({
+			firstName,
+			lastName,
+			creatorType: "contributor",
+			fieldMode: true
+		})
+	})
+	return contributors;
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://www.computerhistory.org/collections/catalog/102658053",
+		"items": [
+			{
+				"itemType": "book",
+				"title": "Knuth, Donald oral history",
+				"creators": [
+					{
+						"firstName": "Edward",
+						"lastName": "Feigenbaum",
+						"creatorType": "contributor",
+						"fieldMode": true
+					},
+					{
+						"firstName": "Donald E.",
+						"lastName": "Knuth",
+						"creatorType": "contributor",
+						"fieldMode": true
+					},
+					{
+						"firstName": "Yan",
+						"lastName": "Rosenshteyn",
+						"creatorType": "contributor",
+						"fieldMode": true
+					}
+				],
+				"date": "2007-03-14; 2007-03-21",
+				"abstractNote": "Collection Title: Oral history collection\nSubject: Fellow Award Honoree; Knuth, Donald; IBM 650 (Computer); Combinatorial analysis--Data processing; Combinatorics; Analysis of algorithms; The Art of Computer Programming; Stanford University; TeX; METAFONT; Religion; Literate programming\nCategory: Transcription\nDescription: In this wide-ranging interview conducted by Edward Feigenbaum, Donald Knuth talks about the progression of his life and career.  Topics include his family background and early interest in music, physics and mathematics, his first exposure to programming,  finding a mentor, and writing a doctoral thesis.   He describes how \"The Art of Computer Programming\" became \"the story of my life\", and why it was put on hold for the TeX and METAFONT projects.  He also talks about personal work habits, programming style, analysis of algorithms, the influence of religion in his life, and his advice to the next generation of scientists.\nType: Document",
+				"archive": "Computer History Museum",
+				"archiveLocation": "X3926.2007",
+				"libraryCatalog": "102658053",
```

Comment:
What is this number? This shouldn't be in `libraryCatalog`.

Reply:
thx. moved to Extra.